app/Http/Controllers/V2/Terrafund/TerrafundCreateGeometryController.php

```diff
@@ -0,0 +1,757 @@
+        }
+        $geojson = DB::selectOne('SELECT ST_AsGeoJSON(geom) AS geojson FROM polygon_geometry WHERE uuid = :uuid', ['uuid' => $uuid])->geojson;
+        $geojsonArray = json_decode($geojson, true);
+        $spikes = $this->detectSpikes($geojsonArray);
+        $SPIKE_CRITERIA_ID = 8;
+        $valid = count($spikes) === 0;
+        $insertionSuccess = $this->insertCriteriaSite($uuid, $SPIKE_CRITERIA_ID, $valid);
+
+        return response()->json(['spikes' => $spikes, 'geometry_id' => $uuid, 'insertion_success' => $insertionSuccess, 'valid' => $valid], 200);
+    }
+
+    public function validatePolygonSize(Request $request)
+    {
+        $uuid = $request->query('uuid');
+        $geometry = PolygonGeometry::where('uuid', $uuid)->first();
+
+        if (! $geometry) {
+            return response()->json(['error' => 'Geometry not found'], 404);
+        }
+
+        $areaSqDegrees = DB::selectOne('SELECT ST_Area(geom) AS area FROM polygon_geometry WHERE uuid = :uuid', ['uuid' => $uuid])->area;
+        $latitude = DB::selectOne('SELECT ST_Y(ST_Centroid(geom)) AS latitude FROM polygon_geometry WHERE uuid = :uuid', ['uuid' => $uuid])->latitude;
+        $areaSqMeters = $areaSqDegrees * pow(111320 * cos(deg2rad($latitude)), 2);
+        $SIZE_CRITERIA_ID = 6;
+        $valid = $areaSqMeters <= 10000000;
+        $insertionSuccess = $this->insertCriteriaSite($uuid, $SIZE_CRITERIA_ID, $valid);
+
+        return response()->json([
+          'area_hectares' => $areaSqMeters / 10000, // Convert to hectares
+          'area_sqmeters' => $areaSqMeters,
+          'geometry_id' => $geometry->id,
+          'insertion_success' => $insertionSuccess,
+          'valid' => $valid,
+        ], 200);
+    }
+
+    public function checkWithinCountry(Request $request)
+    {
+        $polygonUuid = $request->input('uuid');
+
+        if ($polygonUuid === null || $polygonUuid === '') {
+            return response()->json(['error' => 'UUID not provided'], 200);
+        }
+
+        $geometry = PolygonGeometry::where('uuid', $polygonUuid)->first();
+
+        if (! $geometry) {
+            return response()->json(['error' => 'Geometry not found'], 404);
+        }
+
+        $totalArea = DB::table('polygon_geometry')
+            ->where('uuid', $polygonUuid)
+            ->selectRaw('ST_Area(geom) AS area')
+            ->first()->area;
+
+        // Find site_polygon_id and project_id using the polygonUuid
+        $sitePolygonData = SitePolygon::where('poly_id', $polygonUuid)
+            ->select('id', 'project_id')
+            ->first();
+
+        if (! $sitePolygonData) {
+            return response()->json(['error' => 'Site polygon data not found for the specified polygonUuid'], 404);
+        }
+
+        // Find the country ISO using project_id from v2projects
+        $countryIso = DB::table('v2_projects')
+            ->where('uuid', $sitePolygonData->project_id)
+            ->value('country');
```

Comment:
This can easily use the eloquent queries:
```
$countryIso = Project::isUuid($sitePolygonData->project_id)->select('county')->first()->country
```

Even better would be to establish a `belongsTo` relationship on `SitePolygon` -> `Project` (I don't even see `project_id` as a fillable on the model implementation?) so this could be:
```
$countryIso = $sitePolygonData->project()->select('country')->first()->country
```

Reply:
Done the relation between sitePolygon and Project. 
Changed to suggested code